packages/bento-design-system/src/Modal/createModal.tsx

```diff
@@ -86,11 +90,24 @@ export function createModal(
 
     const { defaultMessages } = useDefaultMessages();
 
+    const kind = props.kind == null && props.isDestructive ? "destructive" : props.kind ?? "normal";
```

Comment:
The behavior is to preserve backward compatibility:
- if the user is using `isDestructive` we just look at that
- if the user sets `kind` then we ignore `isDestructive`